thunder/core/transforms.py

```diff
@@ -3733,8 +3733,19 @@ def ones_like(x):
     finally:
         reset_tracectx(tracectx_token)
 
+    saved_for_backward = forward_trace.output[1]
+    flat_saves, _ = tree_flatten(saved_for_backward)
+
     def backward_fn(saved_for_backward, cotangents):
+        # trace converts all saved_for_backward into proxy, we want to restore number scalars afterwards.
+        flat_saves_proxified, saves_spec = tree_flatten(saved_for_backward)
+        flat_filtered = [
+            proxified if isinstance(entry, Proxy) else entry
+            for proxified, entry in zip(flat_saves_proxified, flat_saves)
+        ]
+        saved_for_backward = tree_unflatten(flat_filtered, saves_spec)
```

Comment:
tree_flatten -> map -> tree_unflatten can be handled with a single tree_map call

Reply:
Let me try if tree_flatten would work with zipped trees~~